BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewItemProcessor.swift

```diff
@@ -533,6 +533,7 @@ private extension ViewItemProcessor {
 
                 if case var .data(itemState) = newState.loadingState {
                     itemState.loginState.totpState = totpState
+                    itemState.collections = try await services.vaultRepository.fetchCollections(includeReadOnly: true)
                     newState.loadingState = .data(itemState)
                 }
```

Comment:
🤔 If this throws, does the view state remain ok for the user? I believe it does but wanted to raise it up just in case something is missing given the `totpState` is being updated but not the `loadingState`.

Reply:
The `totpState` won't get updated either, because it'll error out before the updated `itemState` is assigned within `newState` or `state` is updated with `newState`. There are a number of other cases that can error out here before the `state` is updated; the main difference is that the rest of the checks occur earlier in the `do` block, which come to think of it, probably makes sense to be consistent in that regard.